src/data/pipeline-builds/all-pipeline-builds-query.ts

```diff
@@ -0,0 +1,44 @@
+import { apiPaths, createApiPath } from "../api";
+import { UseQueryOptions, useQuery } from "@tanstack/react-query";
+import { FetchError } from "@/lib/fetch-error";
+import { PipelineBuildPage } from "@/types/pipeline-builds";
+export type PipelineBuildOverview = {
+	buildId: string;
+};
+
+export function getPipelineBuildQueryKey({ buildId }: PipelineBuildOverview) {
+	return ["builds", buildId];
+}
+
+export async function fetchAllPipelineBuilds({ buildId }: PipelineBuildOverview, token?: string) {
+	const url = createApiPath(apiPaths.runs.detail(buildId));
+
+	const res = await fetch(url, {
+		method: "GET",
+		credentials: "include",
+		headers: {
+			"Content-Type": "application/json",
+			...(token && { Authorization: `Bearer ${token}` })
+		}
+	});
+
+	if (!res.ok) {
+		throw new FetchError({
+			message: `Error while fetching pipeline run ${buildId}`,
+			status: res.status,
+			statusText: res.statusText
+		});
+	}
+	return res.json();
+}
+
+export function useAllPipelineBuilds(
```

Comment:
I think this is fetching just a specific PipelineBuild, which makes `useAllPipelineBuilds` misleading here as well